### ISSUE TITLE
fitted/predict slow for large models

### ISSUE THREAD

**Issue by bbolker** [glmmTMB/glmmTMB#584]
Hope to get a self-contained example here soon, but: while answering [this question](https://stats.stackexchange.com/questions/465699/analyzing-a-computer-adaptive-test/465864#465864) I noticed that `predict()`/`fitted()`/`residuals()` are very slow.

In `predict()`/`fitted()` the bottleneck is line 313, 

```
newObj$fn(oldPar)  ## call once to update internal structures
```
is there any (even slightly unsafe/optional) way to bypass this?

Do we want to pass `...` through from fitted so that e.g. one could specify `se.fit=FALSE`?  Or is that appropriate in any case?  Same for `residuals.glmmTMB` ?

**Comment by mebrooks** [comment]
Thanks for figuring out the bottleneck. I agree that would be really useful if we could speed it up. I'm using the predict function a ton in bootstrapping some fisheries gear selectivity models lately and it's slow. Do any of you more experienced with TMB have an idea of how to speed it up? @kaskr @casperwberg @fishfollower

**Comment by bbolker** [comment]
This is fixed (see #644 )
